src/blocks/tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,102 @@
+import { type FC, useRef, useState } from 'react';
+import styled from 'styled-components';
+import type { TooltipContainerProps, TooltipProps } from './Tooltip.types';
+import { fadeInAnimation, fadeOutAnimation, getTooltipPositionalCSS, getTooltipResponsiveCSS } from './Tooltip.utils';
+import { getBlocksColor } from 'blocks/Blocks.utils';
+import { useDimensions } from 'hooks/useDimensions';
+
+import { tooltipCSSPropsKeys } from './Tooltip.constants';
+
+const StyledTooltip = styled.div.withConfig({
```

Comment:
For scroll containers the tooltip is not overlapping the container and getting cutoff. So it needs to be a react portal where a new node is created outside the main body tag. The https://reach.tech/tooltip library uses this approach, we can use this.

<img width="736" alt="Screenshot 2024-06-17 at 12 31 11 PM" src="https://github.com/push-protocol/push-dapp/assets/25508878/473f168d-be46-4d36-ad9a-1fc831c06a1a">